src/tests/conformance/post_handshake/validators.rs

```diff
@@ -55,3 +71,255 @@ async fn c015_TM_VALIDATOR_LIST_COLLECTION_node_should_send_validator_list() {
     };
     perform_expected_message_test(Default::default(), &check).await;
 }
+
+fn create_sha512_half_digest(buffer: &Vec<u8>) -> [u8; 32] {
+    let mut hasher = Sha512::new();
+    hasher.update(buffer);
+    let result = hasher.finalize();
+
+    // we return 32 bytes of 64-byte result
+    let mut signature = [0u8; 32];
+    signature.copy_from_slice(&result[..32]);
+    signature
+}
+
+// run this to create a key pair.
+// use the strings to instantiate a SecretKey and PublicKey
+fn _gen_keys() {
+    let engine = Secp256k1::new();
+    let (private_key, public_key) = engine.generate_keypair(&mut secp256k1::rand::thread_rng());
+    let secret_bytes = private_key.secret_bytes();
+    let public_bytes = public_key.serialize();
+    let secret_hex = hex::encode_upper(secret_bytes);
+    let public_hex = hex::encode_upper(public_bytes);
+    println!("secret key hex string {}", secret_hex);
+    println!("public key hex string {}", public_hex);
+}
+
+fn create_validator_blob_json(manifest: &Vec<u8>, pkstr: &str) -> String {
+    let mstr = base64::encode(manifest);
+    let v = Validator {
+        validation_public_key: pkstr.to_owned(),
+        manifest: mstr,
+    };
+    // let mut vvec: Vec<Validator> = Vec::new();
+    // vvec.push(v);
+    let vvec: Vec<Validator> = vec![v];
+
+    // Set expiration to 1 year from now.
+    // validator blob uses delta from Jan 1 2000,
+    // and not 1970, per Unix epoch time
+    let start = SystemTime::now();
+    let epoch = start
+        .duration_since(UNIX_EPOCH)
+        .expect("Time went backwards");
+    let now = epoch.as_secs() as u32;
+    let jan1_2000: u32 = 946684800;
+    let year: u32 = 86400 * 365;
+    let expiration: u32 = now + year - jan1_2000;
+
+    let vblob = ValidatorBlob {
+        sequence: 2022100501,
+        expiration,
+        validators: vvec,
+    };
+    serde_json::to_string(&vblob).unwrap()
+}
+
+fn create_signable_manifest(
+    sequence: u32,
+    public_key: &Vec<u8>,
+    signing_pub_key: &Vec<u8>,
+) -> Vec<u8> {
+    let size = 5 + 2 + public_key.len() + 2 + signing_pub_key.len();
+    let mut manifest: Vec<u8> = vec![0; size];
+    manifest[0] = 0x24;
+    manifest[1] = ((sequence >> 24) & 0xff) as u8;
+    manifest[2] = ((sequence >> 16) & 0xff) as u8;
+    manifest[3] = ((sequence >> 8) & 0xff) as u8;
+    manifest[4] = (sequence & 0xff) as u8;
+    let mut i = 5;
+
+    // serialize public key
+    manifest[i] = 0x71; // field code 1 for "PublicKey"
+    manifest[i + 1] = PUBLIC_KEY_SIZE as u8;
+    i += 2;
+    manifest[i..i + PUBLIC_KEY_SIZE].clone_from_slice(public_key.as_slice());
+    i += PUBLIC_KEY_SIZE;
+
+    // serialize signing public key
+    manifest[i] = 0x73; // field code 3 for "SigningPubKey"
+    manifest[i + 1] = PUBLIC_KEY_SIZE as u8;
+    i += 2;
+    manifest[i..i + PUBLIC_KEY_SIZE].clone_from_slice(signing_pub_key.as_slice());
+    manifest
+}
+
+fn create_final_manifest(
+    sequence: u32,
+    public_key: &Vec<u8>,
```

Comment:
Whenever you pass &Vec<T>, you could pass &[T]. This will save one pointer derefence (and I think clippy should complain about it).